python/cudf/cudf/core/series.py

```diff
@@ -1182,8 +1182,20 @@ def drop_duplicates(self, keep="first", inplace=False):
 
     def _mimic_inplace(self, result, inplace=False):
         if inplace:
-            self._data = result._data
-            self._index = result._index
+            from cudf.core.multiindex import MultiIndex
+
+            self._column._mimic_inplace(result._column, inplace=True)
+            if isinstance(result._index, MultiIndex):
+                for in_col, res_col in zip(
+                    self.index._source_data._columns,
+                    result.index._source_data._columns,
+                ):
+                    in_col._mimic_inplace(res_col, inplace=True)
```

Comment:
Should we just build a `MultiIndex._mimic_inplace` and use that here?

Reply:
Works, will do that.